src/main/java/frc/utils/mirrorutils/Mirrorable.java

```diff
@@ -0,0 +1,47 @@
+package frc.utils.mirrorutils;
+
+import frc.utils.DriverStationUtils;
+
+/**
+ * A class that allows for objects to be mirrored across the center of the field when the robot is on the red alliance.
+ * This is useful for placing field elements and other objects that are mirrored across the field, or for mirroring the target
+ * heading to face a field element.
+ *
+ * @param <T> the type of object to mirror
+ */
+public abstract class Mirrorable<T> {
+
+    protected final T nonMirroredObject, mirroredObject;
+
+    protected final boolean mirrorWhenRedAlliance;
+
+    /**
+     * Creates a new mirrorable object.
+     *
+     * @param nonMirroredObject the object when the robot is on the blue alliance, or the non-mirrored object
+     * @param mirrorWhenRedAlliance whether to mirror the object when the robot is on the red alliance
+     */
+    protected Mirrorable(T nonMirroredObject, boolean mirrorWhenRedAlliance) {
+        this.nonMirroredObject = nonMirroredObject;
+        this.mirroredObject = mirror(nonMirroredObject);
+        this.mirrorWhenRedAlliance = mirrorWhenRedAlliance;
+    }
+
+    /**
+     * @return the current object.
+     * If the robot is on the red alliance and the object should be mirrored, the mirrored object is returned.
+     * Otherwise, the non-mirrored object is returned.
+     */
+    public T get() {
+        return DriverStationUtils.isRedAlliance() && mirrorWhenRedAlliance ? mirroredObject : nonMirroredObject;
```

Comment:
RedAlliance might be problematic here, and in general you have "default" alliance on multiple instances, use the default alliance and check if it is the defualt

Reply:
Default is the one that the sim sets.
I can make constant of basic alliance pose relative or something like this.
Also I will probably delete the mirror package and change it to just do mirror in field, it’s too complicated like it is now

Reply:
חברה לנשום עמוק וללכת לישון

תניח שרז צודק

Reply:
![image](https://github.com/GreenBlitz/GB-Robot-Template/assets/108526263/3728b53e-87e6-41f4-9317-f91c211c160c)

Reply:
Do it